server/api_test.go

```diff
@@ -382,6 +385,77 @@ func TestHandleEndPoll(t *testing.T) {
 	}
 }
 
+func TestPostEndPollAnnouncement(t *testing.T) {
+	api1 := &plugintest.API{}
+	api1.On("GetTeam", "teamID1").Return(&model.Team{Name: "team1"}, nil)
+	api1.On("GetPost", "postID1").Return(&model.Post{ChannelId: "channelID1"}, nil)
+	api1.On("CreatePost", &model.Post{
+		UserId:    "userID1",
+		ChannelId: "channelID1",
+		RootId:    "postID1",
+		Message:   fmt.Sprintf(endPollSuccessfullyFormat, "Question1", "http://localhost:8065/team1/pl/postID1"),
+		Type:      model.POST_DEFAULT,
+		Props: model.StringInterface{
+			"override_username": responseUsername,
+			"override_icon_url": fmt.Sprintf(responseIconURL, "http://localhost:8065", PluginId),
+			"from_webhook":      "true",
+		},
+	}).Return(nil, nil)
+	defer api1.AssertExpectations(t)
+
+	api2 := &plugintest.API{}
+	api2.On("GetTeam", "teamID1").Return(nil, &model.AppError{})
+	api2.On("LogError", GetMockArgumentsWithType("string", 3)...).Return(nil)
+	defer api2.AssertExpectations(t)
+
+	api3 := &plugintest.API{}
+	api3.On("GetTeam", "teamID1").Return(&model.Team{Name: "team1"}, nil)
+	api3.On("GetPost", "postID1").Return(nil, &model.AppError{})
+	api3.On("LogError", GetMockArgumentsWithType("string", 3)...).Return(nil)
+	defer api3.AssertExpectations(t)
+
+	api4 := &plugintest.API{}
+	api4.On("GetTeam", "teamID1").Return(&model.Team{Name: "team1"}, nil)
+	api4.On("GetPost", "postID1").Return(&model.Post{ChannelId: "channelID1"}, nil)
+	api4.On("CreatePost", mock.AnythingOfType("*model.Post")).Return(nil, &model.AppError{})
+	api4.On("LogError", GetMockArgumentsWithType("string", 3)...).Return(nil)
+	defer api4.AssertExpectations(t)
+
+	for name, test := range map[string]struct {
+		API      *plugintest.API
+		Request  *model.PostActionIntegrationRequest
+		Question string
+	}{
+		"Valid request": {
+			API:      api1,
+			Request:  &model.PostActionIntegrationRequest{UserId: "userID1", PostId: "postID1", TeamId: "teamID1"},
+			Question: "Question1",
+		},
+		"Valid request, GetTeam fails": {
+			API:      api2,
+			Request:  &model.PostActionIntegrationRequest{UserId: "userID1", PostId: "postID1", TeamId: "teamID1"},
+			Question: "Question1",
+		},
+		"Valid request, GetPost fails": {
+			API:      api3,
+			Request:  &model.PostActionIntegrationRequest{UserId: "userID1", PostId: "postID1", TeamId: "teamID1"},
+			Question: "Question1",
+		},
+		"Valid request, CreatePost fails": {
+			API:      api4,
+			Request:  &model.PostActionIntegrationRequest{UserId: "userID1", PostId: "postID1", TeamId: "teamID1"},
+			Question: "Question1",
+		},
+	} {
+		t.Run(name, func(t *testing.T) {
+			p := &MatterpollPlugin{
+				ServerConfig: &model.Config{ServiceSettings: model.ServiceSettings{SiteURL: model.NewString("http://localhost:8065")}},
+			}
+			p.SetAPI(test.API)
```

Comment:
I propose to use `setupTestPlugin` here. Its used in every other test and it makes things consistent.

Reply:
:+1: